src/huggingface_hub/repository.py

```diff
@@ -160,32 +164,42 @@ def clone_from(self, repo_url: str, use_auth_token: Union[bool, str, None] = Non
         is_hf_id = len(repo_url.split("/")) <= 2
         api = HfApi()
 
+        if not is_hf_id and repo_url.split("/")[-3] == "datasets":
+            self.repo_type = "dataset"
```

Comment:
You can probably use `REPO_TYPES_URL_PREFIXES` to make this more robust?

maybe implement a `repo_type_and_id_from_hf_id` that takes a 3-length id and returns an optional repo_type, and a 2-length repo_id?

Reply:
Great idea!